### PR TITLE
mariadb: fix musl, make Numa optional; pmdk: remove

### PR DESCRIPTION
**Pull request by superherointj**
MariaDB:
* fix musl
* remove pmdk support
* make Numa optional
  It's only used by large servers. Most users don't have hardware that require Numa. Whoever needs it can use from the flag. Should it be documented? Where (specifically)?

pmdk: remove
  * No further support or maintenance planned by upstream.
     https://pmem.io/blog/2022/11/update-on-pmdk-and-our-long-term-support-strategy/
  * Alpine removed pmdk package:
     https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/43663
  * pmdk is broken for Musl: https://github.com/pmem/pmdk/issues/4157

References:
  - AUR: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=mariadb-git
  - Alpine: https://git.alpinelinux.org/aports/tree/main/mariadb/APKBUILD

Cherry-pick is merged to Haskell:
  * haskellPackages.hslua-core:
    -  #214942

### CODE REVIEW

**Comment by ajs124** [comment]
> * Should it be documented? Where (specifically)?

You could probably put it into the release notes? Idk if it's really worth mentioning though.

**Review by wegank — APPROVED** [review]

**Comment by ajs124** [comment]
Do we have something like "notability guidelines" for the release notes? Something like "MariaDB is built without NUMA support by default now" is IMO enough, because people that would be affected by it understand it, but then again, I don't think "bloated" release notes that nobody reads through are useful, either.

Also, the people that this matters to, if they even exist, probably have monitoring and metrics in place that would alert them, if this actually negatively impacted performance for them.

**Comment by superherointj** [comment]
@ajs124 Can I merge this as it is?

**Comment by ajs124** [comment]
Yeah, should be fine. I ran nixpkgs review and a bunch of things (e.g. amarok) failed to build, but they all seem unrelated.

File: pkgs/servers/sql/mariadb/default.nix [hunk 1]
```diff
@@ -169,16 +173,14 @@ let
...
         ++ lib.optionals stdenv.hostPlatform.isLinux [ linux-pam ]
-        ++ lib.optional (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64) pmdk.dev
         ++ lib.optional (!stdenv.hostPlatform.isDarwin) mytopEnv
         ++ lib.optionals withStorageMroonga [ kytea libsodium msgpack zeromq ]
         ++ lib.optionals (lib.versionAtLeast common.version "10.7") [ fmt_8 ];
 
       propagatedBuildInputs = lib.optionals withEmbedded
-        (lib.optional (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64) pmdk.lib
-         ++ lib.optional (stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAarch32) numactl);
+        (lib.optional withNuma numactl);
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/mariadb/default.nix:183]: The conditions could have been combined here IIUTC

**Pull request merged**